methods/inputs/generate_carbon_density.py

```diff
@@ -15,86 +16,182 @@ def generate_carbon_density(
+        # compute the standard error
+        standard_error = std_dev_agbd / np.sqrt(num_points)
+        # degrees of freedom
+        df = num_points - 1
+        # t-critical value for 95% confidence interval
+        t_critical = stats.t.ppf(0.975, df)  # for 95% confidence
+        # margin of error
+        margin_of_error = t_critical * standard_error
+        # compute confidence interval for mean agbd
+        ci_lower_agbd = mean_agbd - margin_of_error
+        ci_upper_agbd = mean_agbd + margin_of_error
+
+        # convert median agbd to carbon density
+        bgbd_median = median_agbd * 0.2  # below-ground biomass density
+        deadwood_bd_median = median_agbd * 0.11  # deadwood biomass density
+        total_median = median_agbd + bgbd_median + deadwood_bd_median
+        carbon_density_median = total_median * 0.47  # convert biomass to carbon density
+
+        # convert lower quartile agbd to carbon density
+        bgbd_lower = lower_quartile_agbd * 0.2
+        deadwood_bd_lower = lower_quartile_agbd * 0.11
+        total_lower = lower_quartile_agbd + bgbd_lower + deadwood_bd_lower
+        carbon_density_lower = total_lower * 0.47
+
+        # convert upper quartile agbd to carbon density
+        bgbd_upper = upper_quartile_agbd * 0.2
+        deadwood_bd_upper = upper_quartile_agbd * 0.11
+        total_upper = upper_quartile_agbd + bgbd_upper + deadwood_bd_upper
+        carbon_density_upper = total_upper * 0.47
+
+        # convert mean agbd to carbon density
+        bgbd_mean = mean_agbd * 0.2
+        deadwood_bd_mean = mean_agbd * 0.11
+        total_mean = mean_agbd + bgbd_mean + deadwood_bd_mean
+        carbon_density_mean = total_mean * 0.47
+
+        # convert confidence interval bounds to carbon density
+        bgbd_ci_lower = ci_lower_agbd * 0.2
+        deadwood_bd_ci_lower = ci_lower_agbd * 0.11
+        total_ci_lower = ci_lower_agbd + bgbd_ci_lower + deadwood_bd_ci_lower
+        carbon_density_ci_lower = total_ci_lower * 0.47
+
+        bgbd_ci_upper = ci_upper_agbd * 0.2
+        deadwood_bd_ci_upper = ci_upper_agbd * 0.11
+        total_ci_upper = ci_upper_agbd + bgbd_ci_upper + deadwood_bd_ci_upper
+        carbon_density_ci_upper = total_ci_upper * 0.47
+
+        # compute standard deviation of carbon density
+        # since the conversion factors are linear, we can apply the same factors
+        # total biomass multiplier: 1 + 0.2 + 0.11 = 1.31
+        # carbon density multiplier: 1.31 * 0.47 = 0.6157
+        std_dev_carbon_density = std_dev_agbd * 0.6157
+
+        # append the results to the list
+        results.append([
+            land_use_class,
+            carbon_density_median,
+            carbon_density_lower,
+            carbon_density_upper,
+            carbon_density_mean,
+            carbon_density_ci_lower,
+            carbon_density_ci_upper,
+            std_dev_carbon_density,
+            num_points
```

Comment:
This is a methodological change, so what is the plan to update the methodology document for the next release? Can I confirm that someone has this task?

Reply:
Yes sorry, this was mainly a later addition so I could run side experiments on the influence of changing the GEDI year range and the buffer size upon the resulting carbon densities. I'll remove it from the code so that there is no proposed change to the existing `generate_carbon_density.py` script.